### PR TITLE
[RISCV] Lower vfmv.s.f intrinsics to VFMV_S_F_VL first

### PR DESCRIPTION
**Pull request by lukel97**
Currently vfmv.s.f intrinsics are directly selected to their pseudos via a
tablegen pattern in RISCVInstrInfoVPseudos.td, whereas the other move
instructions (vmv.s.x/vmv.v.x/vmv.v.f etc.) first get lowered to their
corresponding VL SDNode, then get selected from a pattern in
RISCVInstrInfoVVLPatterns.td

This patch brings vfmv.s.f inline with the other move instructions, and shows
how the LMUL reducing combine for VFMV_S_F_VL and VMV_S_X_VL results in the
discrepancy in the test added in #75544.

Split out from #71501, where we did this to preserve the behaviour of selecting
vmv_s_x for VFMV_S_F_VL for small enough immediates.

### CODE REVIEW

File: llvm/test/CodeGen/RISCV/rvv/vsetvli-insert.ll [hunk 1]
```diff
@@ -630,8 +630,9 @@ define void @add_v16i64(ptr %x, ptr %y) vscale_range(2,2) {
 define <vscale x 2 x float> @fp_reduction_vfmv_s_f(float %0, <vscale x 8 x float> %1, i64  %2) {
 ; CHECK-LABEL: fp_reduction_vfmv_s_f:
 ; CHECK:       # %bb.0:
-; CHECK-NEXT:    vsetvli zero, a0, e32, m4, ta, ma
+; CHECK-NEXT:    vsetvli zero, a0, e32, m1, ta, ma
 ; CHECK-NEXT:    vfmv.s.f v12, fa0
+; CHECK-NEXT:    vsetvli zero, a0, e32, m4, ta, ma
```
Comment (topperc) [inline llvm/test/CodeGen/RISCV/rvv/vsetvli-insert.ll:635]: This is a regression. Do you have a plan to fix this?

Comment (lukel97) [reply llvm/test/CodeGen/RISCV/rvv/vsetvli-insert.ll:635]: I think @preames's [draft patch here](https://github.com/llvm/llvm-project/pull/75544#issuecomment-1858133214) should fix it. Happy to take over it if there's no objections

Comment (lukel97) [reply llvm/test/CodeGen/RISCV/rvv/vsetvli-insert.ll:635]: https://github.com/llvm/llvm-project/pull/76801

**Review by preames — CHANGES_REQUESTED** [review]
Needs rebased after https://github.com/llvm/llvm-project/pull/76801, marking as request changes to reflect and make it easier to scan review list for actionable items.

**Review by topperc — APPROVED** [review]
LGTM

**Pull request merged**